src/google/protobuf/port_def.inc

```diff
@@ -505,7 +505,11 @@
 #define PROTOBUF_ALIGNAS(byte_alignment) alignas(byte_alignment)
 #endif
 
+#ifndef GOOGLE_PROTOBUF_NO_FINAL
```

Comment:
Unfortunately I don't think we will want to accept this change relating to the `final` keyword. It's important for generated messages to be final, because we need to be able to change things without worrying about breaking derived classes. Marking messages final also allows the compiler to generate better code by devirtualizing some member function calls.

Reply:
Both valid points but this flag could be documented in a way that let the users know that they can use it at their own risk and that Google does not guarantee the backward / forward compatibility between releases. As this flag would not be defined by default, only users that would choose to do use derivation would do it knowing the potential impacts and in this case we let the final decision in the user hands. 

In our case the from the protobuf generated object because we add a layer that adds the capability to create observers to be notified when protobuf objects are received from the network. We could also use composition instead of derivation but derivation was giving us a cleaner / more direct  access to the underlying protobuf object and its attributes.

Reply:
I'm sorry but I checked with some other members of the protobuf team and the consensus is that we do not want to support inheriting from generated messages. If you really need to inherit from generated messages and can't easily refactor to avoid that, you can always make a small modification to port_def.inc in your local copy of the source.

Reply:
No problem, thanks for looking into this with the rest of the team. I understand your reasons, they are valid ones! Modifying the .h is what we were doing so far but we were trying to avoid doing this minor change. We'll see what we do with this, maybe we continue this way or maybe I will simply change the object to use composition of derivation. I will roll back this change.